boot/LoadLinux.c

```diff
@@ -21,15 +21,16 @@
 #include "BootParser.h"
 #include "config.h"
 #include "iso_fs.h"
+#include "lib/misc/LED.h"
 
 //Grub bits
 unsigned long saved_drive;
 grub_error_t errnum;
 unsigned long saved_partition;
 unsigned long boot_drive;
-	
+
 static int nRet;
-static u32 dwKernelSize = 0, dwInitrdSize = 0;
+static s32 dwKernelSize = 0, dwInitrdSize = 0;
```

Comment:
Why make these size values signed?

Reply:
It's a crash fix. The functions to read files from a filesystem return the size, and -1 indicates that the file wasn't found. But with `dwKernelSize` being unsigned, the `< 0` check always failed even when the file wasn't found, so instead of printing the error message, Cromwell tried to memcpy `0xffffffff` bytes and subsequently crashed.